### PR TITLE
Ratio Component Refactor + Jest Testing Coverage

### PR DESCRIPTION
**Pull request by sghoweri**
## Jira
http://vjira2:8080/browse/BDS-987

## Summary
- Refreshes the ratio component's internals + adds testing coverage support via Jest (including visual regression testing)

image
ratio-js-bolt-ratio-component-bolt-ratio-with-html-5-video-renders-1-snap
> Screenshot snapshots for VRT in the `<bolt-ratio>` component

## Details
- Cleans up and refactors the ratio component to simplify rendering logic + allow for more easily testing the Shadow DOM-related rendering behavior + conditional behavior based on support for CSS Custom properties  
- Adds several new Jest tests to the Ratio component — checking the ratio component’s Twig rendering and the web component rendering!
- Also adds two new visual regression tests to the ratio component (one with a vanilla HTML5 `<video>` example and the other with a vanilla `` tag example)
- Updates then `.boltrc` config + Jest-related logic to copy over the example video asset used for demos & testing
- Updates the Jest test fixture logic to auto-copy over any non-image assets (previously these were not copied over)
  - Fixes an issue where the Jest testing environment didn’t have the Bolt webfonts correctly copied over + wasn’t allowed to access these assets when being served from the `www` folder
  - This should also address previous VRT tests / test differences noticed with the button component
- Tests out adding a few component-specific examples to the component's schema file directly (potential emerging pattern on how we can handle demoing some of these examples automatically)
- Updates the `<bolt-ratio>` component schema to differentiate between Web Component vs Twig-specific props
- Adjusted the docs site schema table logic to better differentiate between the deprecated vs non-deprecated prop options

## Deprecated
The ratio component's `aspect-ratio-height` and `aspect-ratio-width` props are deprecated and will be removed in Bolt v3.0.

## How to test
- Confirm all Travis CI build tests passing
- Spot check other component examples and page prototypes to verify consistent behavior / styling

### CODE REVIEW

File: docs-site/src/pages/pattern-lab/_patterns/02-components/ratio/00-ratio-docs.twig [hunk 1]
```diff
@@ -0,0 +1,18 @@
+{% set usage %}{% verbatim %}
+{% include "@bolt-components-ratio/ratio.twig" with {
+  content: '&lt;img src="https://via.placeholder.com/350x150"&gt;',
+  ratio: "350/150"
+} only %}
+{% endverbatim %}{% endset %}
+
+{# {% include "@utils/docs.twig" with {
```
Comment (adamszalapski) [inline docs-site/src/pages/pattern-lab/_patterns/02-components/ratio/00-ratio-docs.twig]: This is only my suggestion but maybe we should remove commented not used/needed code to our code be more readable and clean. In this component we not have schema and docs for it, that is correct?

Comment (sghoweri) [reply docs-site/src/pages/pattern-lab/_patterns/02-components/ratio/00-ratio-docs.twig]: Maybe add a todo in a comment to re-enable this after a little bit more component explorer work is done?

Ideally we’d be including this block of code for every one of our component docs (like Button and Navbar). 

For this one in particular, I just found that this component didn’t demo as well as the others (without adjusting how the form for this shows up to make it a lot easier and more helpful).

Feel free to uncomment this out locally and view the main ratio component docs in PL to see for yourself 🙂

Comment (sghoweri) [reply docs-site/src/pages/pattern-lab/_patterns/02-components/ratio/00-ratio-docs.twig]: But yes, normally I’d recommend we don’t include commented out code we didn’t intend to ship — this one thing just needs a little more work first.

Comment (adamszalapski) [reply docs-site/src/pages/pattern-lab/_patterns/02-components/ratio/00-ratio-docs.twig]: Thanks for the answer. I have only such a reservation all other looks great. I think yes the todo comment will be good information that work is in progress. And when i check some components i see that we update something we don't clean unused code, we must change that (not in this situation). I accept this PR fully.

**Comment by sghoweri** [comment]
Thanks @adamszalapski!

**Pull request merged**